python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -0,0 +1,464 @@
+        result_series.index = index_prefixes
+
+        # Append the event information to the Series
+        event_dict = vars(self.event)
+        event_dict = {
+            key: val
+            for key, val in event_dict.items()
+            if key in ["dayObs", "seqNum", "version"]
+        }
+
+        # Create a pandas Series from the dictionary
+        event_series = pd.Series(event_dict)
+
+        # Concatenate the two Series
+        result_series = pd.concat([event_series, result_series])
+
+        # Display the resulting Series
+        return result_series
+
+
+def find_adjacent_true_regions(
+    series: pd.Series, min_adjacent: None | int = None
+) -> list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]:
+    """
+    Find regions in a boolean Series containing adjacent True values.
+
+    Parameters
+    ----------
+    series : pd.Series
+        The boolean Series to search for regions.
+
+    min_adjacent : int, optional
+        Minimum number of adjacent True values in a region.
+        Defaults to half size of the series.
+
+    Returns
+    -------
+    list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]
+        A list of tuples representing the start and end indices of regions
+        containing more than or equal to min_adjacent adjacent True values.
+    """
+    min_adjacent = min_adjacent if min_adjacent else 0.5 * series.size
+    regions = []
+    for key, group in series.groupby((series != series.shift()).cumsum()):
+        if key and len(group) >= min_adjacent:
+            region_indices = group.index
+            regions.append((region_indices.min(), region_indices.max()))
+    return regions
+
+
+def get_tma_slew_event(day_obs: int, seq_number: int) -> TMAEvent:
+    """
+    Retrieve Telescope Mount Assembly (TMA) slew events within a specified time
+    range.
+
+    Parameters
+    ----------
+    dayObs : int
+        Observation day in the YYYYMMDD format.
+    seqNum : int
+        Sequence number associated with the slew event.
+
+    Returns
+    -------
+    lsst.summit.utils.tmaUtils.TMAEvent
+        A TMA slew events that occurred within the specified time range.
+
+    Notes
+    -----
+    This function retrieves TMA slew events occurring between the specified
+    start and end times. It uses the TMAEventMaker class to obtain events for
+    the specified day of observation (dayObs). The events are filtered to
+    include only those that start after 1 second before the specified start
+    time and end before 1 second after the specified end time.
+
+    Returs
+    ------
+    lsst.summit.utils.tmaUtils.TMAEvent
+        A TMA slew event that occurred within the specified time range.
+
+    Raises
+    ------
+    ValueError
+        If no events are found for the specified time range.
+    ValueError
+        If more than one event is found for the specified time range.
+    """
+    logger.info(f"Query events in {day_obs}")
+    events = event_maker.getEvents(day_obs)
```

Comment:
`event_maker` appears to be an undefined local variable, so I suspect this was copied from a notebook, and only runs correctly in one where that's defined. Also, this should therefore be a linting error, so does this mean the linting is broken here?

Reply:
It is defined at the beginning of the file [inertia_compensation_system.py#L26](https://github.com/lsst-sitcom/summit_testing_analysis/blob/05a37ab8a8c2ff7bf008a97a1c5c133abe2b58ef/python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py#L26). If I recall correctly, you told me to avoid instantiating it many times. Another option would be to move it to the `ICSAnalysis.__init__` method. What do you think?

Reply:
You could make `event_maker` a class attribute and pass it in in the` __init__`? something like:

```
  def __init__(
        self,
        event_maker: TMAEventMaker,
        event: TMAEvent,
        inner_pad: float = 0.0,
        outer_pad: float = 0.0,
        n_sigma: float = 1.0,
        logger: logging.Logger | None = None,
    ):

    self.event_maker = event_maker
    ....
```

?

Reply:
The idea was to make it once for a given day, because in order to make the events, it _has_ to query the data for the TMA states for the whole day, so it's not all that cheap, or light on EFD usage, so I encouraged Bruno not to make a fresh one for each event, but rather pass a pre-made one around, so I think this was because of that. See discussion in the SPA room about this though - we'll work it out one way or another, but I don't think adding it to this class, which looks like it is instantiated often, as I see it, will be an improvement.

Reply:
(I mean, _anything_ is an improvement over making one on module import, but I don't think that putting it here is a good option 😉)